docs/USAGE-GUIDE.md

```diff
@@ -1585,6 +1585,102 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config, const uint8_t *n
+
+**s2n_connection_append_psk** appends a pre-shared key object to the list of pre-shared keys supported by the s2n connection. If a pre-shared key with a duplicate identity is found, an error is returned and the pre-shared key is not added to the list. Note that the PSK object is copied to the connection and the user is responsbile for freeing the memory associated with the s2n_psk object. Use `s2n_psk_free` to free the memory associated with the external pre-shared key object after using this API.
+
+**s2n_config_set_psk_mode** sets the PSK mode on the s2n config object. The enum **s2n_psk_mode** lists the supported PSK modes. 
+
+**s2n_connection_set_psk_mode** sets the pre-shared key mode on the s2n connection object. The supported pre-shared key modes are listed in the enum `s2n_psk_mode`. This API overrides the pre-shared key mode for the connection and will continue to use the existing pre-shared key mode even when a new config is set for the connection. 
+
+### Obtaining Negotiated Pre-Shared Keys
+
+The folllowing APIs enable the caller to obtain the pre-shared key that was selected by the server for the connection. 
+
+**s2n_connection_get_negotiated_psk_identity_length** gets the negotiated pre-shared key identity length from the s2n connection object. If the negotiated pre-shared key does not exist, the value `0` is returned.  
+
+**s2n_connection_get_negotiated_psk_identity** gets the negotiated pre-shared key identity from the s2n connection object. If the negotiated pre-shared key does not exist, the PSK identity will not be obtained and no error will be returned.  
+
+### Configuring Client Offered Pre-Shared Keys
+ 
+The folllowing APIs enable the client to configure the list of pre-shared keys supported.
+
+**s2n_offered_psk** creates a new s2n offered pre-shared key object. An offered pre-shared key object represents a single pre-shared key (PSK) shared by the client. This object is used to create the list of offered pre-shared keys shared by the client. Use `s2n_offered_psk_free` to free the memory allocated to the s2n offered pre-shared key object created by this API. 
+
+**s2n_offered_psk_free** frees the memory associated with the `s2n_offered_psk` object.
+
+### Using the Offered Pre-Shared Keys to Select a Pre-Shared Key
+
+The following APIs enable the client to use it's list of offered pre-shared keys to set the callback that selects the matching server pre-shared key to be used for the connection. Use the `s2n_config_set_psk_selection_callback` only when a custom pre-shared key selection logic is desired by the client. By default in the pre-shared key exension, s2n-tls server chooses the first identity in its pre-shared key list that also appears in the client's pre-shared key list.
```

Comment:
These are called by the server, not the client. We also don't need to mention the extension.

Reply:
The offered PSKs are set by the client and its processed by the server.